src/main/java/gift/service/product/option/ProductOptionServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package gift.service.product.option;
+
+import gift.dto.product.option.OptionResponse;
+import gift.entity.product.Product;
+import gift.entity.product.option.ProductOption;
+import gift.entity.product.value.ProductId;
+import gift.exception.custom.OptionAlreadyExistException;
+import gift.exception.custom.OptionNotFoundException;
+import gift.exception.custom.ProductNotFoundException;
+import gift.repository.product.ProductRepository;
+import gift.repository.product.option.ProductOptionRepository;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional
+public class ProductOptionServiceImpl implements ProductOptionService {
+
+    private final ProductRepository productRepository;
+    private final ProductOptionRepository optionRepository;
+
+    public ProductOptionServiceImpl(
+            ProductRepository productRepository,
+            ProductOptionRepository optionRepository) {
+        this.productRepository = productRepository;
+        this.optionRepository = optionRepository;
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public List<OptionResponse> getOptions(Long productId) {
+        productRepository.findById(new ProductId(productId))
+                .orElseThrow(() -> new ProductNotFoundException(productId));
+        return optionRepository.findAllByProduct_Id(new ProductId(productId)).stream()
+                .map(ProductOption::toResponse)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public OptionResponse addOption(Long productId, String name, int quantity) {
+        Product product = productRepository.findById(new ProductId(productId))
+                .orElseThrow(() -> new ProductNotFoundException(productId));
+
+        if (
+                optionRepository.existsByProduct_IdAndName_Name(
+                        new ProductId(productId),
+                        name
+                )
+        ) {
+            throw new OptionAlreadyExistException();
+        }
+
+        ProductOption option = ProductOption.of(name, quantity);
```

Comment:
```suggestion
        ProductOption option = ProductOption.of(name, quantity, product);
```
이렇게 생성하지않고 따로 메서드를 뺀 이유가 있으실까요?

Reply:
객체 생성과 테이블 관계 설정은 다른 영역이라고 생각해서,  조금 더 명시적으로 두기 위해 메서드를 분리했던것으로 기억합니다 !

Reply:
테스트 코드를 위해 기존 메서드도 남겨두고 오버로딩 진행했습니다!

Reply:
```
public void assignTo(Product product) {
        this.product = product;
    }
```
해당 메서드는 이제 테스트에서만 사용하게 되었군요.! 
테스트코드에서만 사용하는 코드를 프로덕션 코드가 가지고 있는 것에 대해서 어떻게 생각하실까요?🙄